donut/modules/directory_search/helpers.py

```diff
@@ -0,0 +1,248 @@
+            NATURAL LEFT JOIN buildings
+            NATURAL LEFT JOIN users
+            NATURAL LEFT JOIN images
+    """
+    query += ' WHERE INSTR(email, %s) > 0'
+    substitution_arguments = [kwargs['email'].lower()]
+    if kwargs['username']:
+        query += ' AND INSTR(username, %s) > 0'
+        substitution_arguments.append(kwargs['username'])
+    if kwargs['name']:
+        name_search, name_query = make_name_query(kwargs['name'])
+        query += ' AND ' + name_query
+        substitution_arguments += name_search
+    if kwargs['house_id']:
+        query += ' AND house.group_id = %s'
+        substitution_arguments.append(kwargs['house_id'])
+    if kwargs['option_id']:
+        query += ' AND option_id = %s'
+        substitution_arguments.append(kwargs['option_id'])
+    if kwargs['building_id']:
+        query += ' AND building_id = %s'
+        substitution_arguments.append(kwargs['building_id'])
+    if kwargs['grad_year']:
+        query += ' AND graduation_year = %s'
+        substitution_arguments.append(kwargs['grad_year'])
+    if kwargs['state']:
+        query += ' AND state = %s'
+        substitution_arguments.append(kwargs['state'])
+    query += ' ORDER BY LOWER(last_name), LOWER(full_name)'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, substitution_arguments)
+        return cursor.fetchall()
+
+
+def members_unique_values(field):
+    query = 'SELECT DISTINCT ' + field + ' FROM members WHERE ' + field + ' IS NOT NULL AND ' + field + '!= "" ORDER BY ' + field
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query)
+        return [member[field] for member in cursor.fetchall()]
+
+
+def get_houses():
+    query = 'SELECT * FROM group_houses ORDER BY group_name'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query)
+        return cursor.fetchall()
+
+
+def get_options():
+    query = 'SELECT * FROM options ORDER BY option_name'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query)
+        return cursor.fetchall()
+
+
+def get_residences():
+    query = 'SELECT DISTINCT building_id, building_name FROM members NATURAL JOIN buildings ORDER BY building_name'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query)
+        return cursor.fetchall()
+
+
+def get_grad_years():
+    return members_unique_values('graduation_year')
+
+
+def get_states():
+    return members_unique_values('state')
+
+
+def get_preferred_name(user_id):
+    query = 'SELECT preferred_name FROM members WHERE user_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, [user_id])
+        return cursor.fetchone()['preferred_name'] or ''
+
+
+def set_preferred_name(user_id, name):
+    query = 'UPDATE members SET preferred_name = %s WHERE user_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, [name, user_id])
+
+
+def get_gender(user_id):
```

Comment:
Move these helper functions and route endpoints over to core.
Some functions which query member are a rehash of the get_member_data helper in core and should use that instead.